packages/bruno-electron/src/app/watcher.js

```diff
@@ -457,14 +456,31 @@ class Watcher {
         depth: 20
       });
 
+      let startedNewWatcher = false;
       watcher
         .on('add', (pathname) => add(win, pathname, collectionUid, watchPath))
         .on('addDir', (pathname) => addDirectory(win, pathname, collectionUid, watchPath))
         .on('change', (pathname) => change(win, pathname, collectionUid, watchPath))
         .on('unlink', (pathname) => unlink(win, pathname, collectionUid, watchPath))
-        .on('unlinkDir', (pathname) => unlinkDir(win, pathname, collectionUid, watchPath));
-
-      self.watchers[watchPath] = watcher;
+        .on('unlinkDir', (pathname) => unlinkDir(win, pathname, collectionUid, watchPath))
+        .on('error', (error) => {
+          if (error.code === 'ENOSPC' && !startedNewWatcher) {
```

Comment:
The On `error` callback is triggered for every file that could not be watched, because of the `ENOSPC` error. So its triggered possible a few times. The `startedNewWatcher` variable is to prevent multiple watcher from being started. I think this should be left in place.

But adding `&& !forcePolling` to the if condition is a good point to prevent a possible endless loop.

I will update the PR later, Thanks for the Review